x/globalfee/ante/fee.go

```diff
@@ -73,6 +73,11 @@ func (mfd FeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, ne
 		return ctx, err
 	}
 
+	// reject the transaction early if the feeCoins have more denoms than the fee requirement
+	if feeCoins.Len() > requiredGlobalFees.Len() {
+		return ctx, sdkerrors.Wrapf(sdkerrors.ErrInvalidCoins, "fee is not a subset of required fees; got %s, required: %s", feeCoins, requiredGlobalFees)
```

Comment:
I think you need to use `.String()` for both `feeCoins` and `requiredGlobalFees`